components/constellation/constellation.rs

```diff
@@ -775,6 +775,27 @@ where
         self.embedder_proxy.send(EmbedderMsg::ShutdownComplete);
     }
 
+    /// Helper that sends a message to the event loop of a given pipeline.
+    /// Returns `false` after printing a warning if the pipeline does not exist.
+    fn send_handling_error(
+        &mut self,
+        pipeline_id: PipelineId,
+        msg: ScriptThreadMessage,
+        warn_msg: &str,
+    ) -> bool {
+        let result = match self.pipelines.get(&pipeline_id) {
+            Some(pipeline) => pipeline.event_loop.send(msg),
+            None => {
+                warn!("{}: {}", pipeline_id, warn_msg);
```

Comment:
As this change is about cleaning up and modernizing the code, can you please put the variable inline into the format strings now that Rust supports that? Here and elsewhere in the change.

```suggestion
                warn!("{pipeline_id}: {pipeline_missing_message}");
```